src/serializer/functions.js

```diff
@@ -239,17 +239,24 @@ export class Functions {
     // check that functions are independent
     let conflicts: Map<BabelNodeSourceLocation, CompilerDiagnostic> = new Map();
     for (let [fun1, call1] of calls) {
+      let fun1Name = this.functionExpressions.get(fun1) || fun1.intrinsicName || "(unknown function)";
       // Also do argument validation here
       let funcLength = fun1.getLength();
       if (funcLength && funcLength > 0) {
         // TODO #987: Make Additional Functions work with arguments
-        throw new FatalError("TODO: implement arguments to additional functions");
+        let error = new CompilerDiagnostic(
+          `Additional function ${fun1Name} has parameters which is not yet supported`,
+          undefined, // TODO: a location would be really useful here
```

Comment:
In general, if a specific location is not available, we default to realm.currentLocation. This will usually identify an expression that is part of the overall construct. In this particular case, the realm's current location has very little to do with what is going on here. The best we can do is to point the programmer to the function itself. fun1.expressionLocation should contain the relevant location.